Libs/vtkSegmentationCore/vtkOrientedImageDataResample.cxx

```diff
@@ -1835,9 +1835,100 @@ bool vtkOrientedImageDataResample::IsLabelInMask(
   return valueFound;
 }
 
+//----------------------------------------------------------------------------
+bool vtkOrientedImageDataResample::ValidateSegmentationImageType(vtkImageData* image)
+{
+  if (!image)
+  {
+    vtkErrorWithObjectMacro(nullptr, "ValidateSegmentationImageType: Invalid input image");
+    return false;
+  }
+
+  vtkIdType validScalarTypes[3] =
+  {
+    VTK_UNSIGNED_CHAR,
+    VTK_UNSIGNED_SHORT,
+    VTK_UNSIGNED_INT
+    //VTK_UNSIGNED_LONG // See issue #5427.
+                        // vtkImageThreshold is not able to fill thresholded images with VTK_LONG_MAX, preventing segment erase.
+  };
+
+  int scalarType = image->GetScalarType();
+  for (int i = 0; i < 3; i++)
```

Comment:
It would be better not to have the hard-coded number 3 here.  Probably an `#define` would be better.

Reply:
Actually, since validScalarTypes is defined locally it probably make sense to use:
```
for (vtkIdType validScalarType : validScalarTypes)
```

Unless we should define the list of valid scalar types elsewhere.